contrib/asmap/asmap-tool.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) 2022 Pieter Wuille
+# Distributed under the MIT software license, see the accompanying
+# file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+
+import argparse
+import sys
+import ipaddress
+import math
+
+import asmap
+
+def load_file(input_file, state=None):
+    try:
+        contents = input_file.read()
+    except OSError as err:
+        sys.exit("Input file '%s' cannot be read: %s." % (input_file.name, err.strerror))
```

Comment:
In fde0193e687ad50a01a191e14fb6c052b3534bc1: perhaps using f-strings in all similar cases?

Reply:
done